purview/utils/general.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import re
+import logging
+from datetime import datetime
+
+log = logging.getLogger(__name__)
+
+
+def timestamp_to_datetime(timestamp):
+    """
+    Parses a timestamp and return a datetime object
+    :param timestamp: a string of a timestamp
+    :return: a datetime object
+    """
+    log.debug('Trying to parse the timestamp "{0}"'.format(timestamp))
+    error_msg = 'The timestamp "{0}" is an invalid format'.format(timestamp)
+    combinations = (
+        (r'^(?P<datetime>\d{4}-\d{1,2}-\d{1,2} \d{1,2}:\d{1,2}:\d{1,2})(?:\.\d+)?$',
+         '%Y-%m-%d %H:%M:%S'),
+        (r'^(?P<datetime>\d{4}-\d{1,2}-\d{1,2})$', '%Y-%m-%d'),
+        (r'^(?P<datetime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})(?:\.\d+)?(?:Z|[-+]00(?::00)?)?$',
```

Comment:
Can we do the same here as well? Or am I missing something?

Reply:
Not there because I'm following the ISO-8601 spec there.

Reply:
ah, sounds good!